igor-web/src/main/java/com/netflix/spinnaker/igor/artifacts/ArtifactController.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.igor.artifacts;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * Controller for supplying version options and artifacts for a trigger that triggers from artifact events.
+ */
+@RestController
+@RequestMapping("/artifacts")
+public class ArtifactController {
+
+  ArtifactServices artifactServices;
+
+  @Autowired
```

Comment:
`@Autowired` is implied on single non-default constructors.

Reply:
I really dislike magic `@Autowired` (when it is implied like you mention). My strong preference is for a clearly annotated constructor for clarity. In general I don't mind a couple more lines for something that clearly indicates what's going on